### PR TITLE
Refactoring: use PrefixedBucketClient in UserBucketClient

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
This is just an internal refactoring PR to use `PrefixedBucketClient` in `UserBucketClient`.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storage/bucket/prefixed_bucket_client.go [hunk 1]
```diff
@@ -23,7 +22,7 @@ func NewPrefixedBucketClient(bucket objstore.Bucket, prefix string) *PrefixedBuc
 }
 
 func (b *PrefixedBucketClient) fullName(name string) string {
-	return path.Join(b.prefix, name)
+	return b.prefix + objstore.DirDelim + name
```
Comment (pracucci) [inline pkg/storage/bucket/prefixed_bucket_client.go:25]: To keep the same exact logic we had in `UserBucketClient`. If name is "", the delimiter must be added anyway.

Comment (jtlisi) [reply pkg/storage/bucket/prefixed_bucket_client.go:25]: I see, sorry I made this change.

**Review by jtlisi — APPROVED** [review]
LGTM

File: pkg/storage/bucket/prefixed_bucket_client.go (hunk 1 above)
Comment (pstibrany) [reply pkg/storage/bucket/prefixed_bucket_client.go:25]: Based on my suggestion, so sorry about that :) 

I'm not sure empty name should be a valid choice though and I wonder how do object stores work in that case.

Comment (pracucci) [reply pkg/storage/bucket/prefixed_bucket_client.go:25]: > I'm not sure empty name should be a valid choice
It's used to list the root. Thanos does it.

Comment (pstibrany) [reply pkg/storage/bucket/prefixed_bucket_client.go:25]: Ah right, I misunderstood and was thinking about empty prefix.

**Review by pstibrany — APPROVED** [review]

**Pull request merged**